### ISSUE TITLE
It is easier to `brew edit` than it is to undo the results of `brew edit`

### ISSUE THREAD

**Issue by mcclure** [Homebrew/brew#5108]
## Reproduction steps

There is a formula (opencv3, if it matters) that I wanted to install with a different set of configuration options from the one Homebrew normally uses. I think, isn't there some way to create a local version of a formula? I do some google searching and I find `brew edit`. Awesome! I `brew edit` and I alter the formula. It worked! That was so easy! I `brew install` the formula. It builds with my configuration options. Everything's great. I run the program. **It's broken. I broke it.** It's completely broken and it's my fault.

I try to undo what I did. I type "how to reverse brew edit" into Google. It finds [a stack overflow thread](https://stackoverflow.com/questions/9369519/reset-homebrew-formula). The stack overflow thread (1) suggests using git, which I don't really know how to use, and (2) gives specific instructions which do not work.

Now I'm stuck with a broken opencv3 and I don't know how to fix what I broke.

<sub>(I eventually fixed it, but that's not the point.)<sub>

##  Bug

I think that this is a user interface bug. `brew edit` is there, enticing, and available to Brew non-experts. Reversing it requires specialized knowledge. Homebrew is sort of encouraging me here to break my install in a way I will not be able to fix.

## Expected behavior

Either
1. It should be harder to `brew edit`,
2. It should be easier to reverse a `brew edit`, 
3. Or running `brew edit` should come with either a warning that what you're about to do may be irreversible, or instructions on how to reverse it.

My preferred solution would be if there were a `brew edit --revert` that did whatever magic git thing reverses a `brew edit`. What's great about this solution is that it would naturally be documented in `brew edit --help`, so there is a natural breadcrumb trail that a nonexpert user could use to find this feature even with no additional messaging. If you broke something with `brew edit` the chances are high that you would look in the help for `brew edit` while trying to figure out how to undo this.

## Environment

```
$ brew -v
Homebrew 1.7.7-53-ga2074f0
Homebrew/homebrew-core (git revision 6ada8; last commit 2018-10-12)
Homebrew/homebrew-cask (git revision d1662; last commit 2018-10-13)
```

**Comment by colindean** [comment]
This rough edge is why I've kept a parallel clone of homebrew-core for making edits to formulae and adding new ones.

Maybe there's a (documented?) workflow that I and others have been missing…

**Comment by MikeMcQuaid** [comment]
@mcclure Sorry for the pain you've experienced! `brew update-reset` can be used to revert to the upstream state on all Homebrew's repositories.

**Comment by colindean** [comment]
> Maybe there's a (documented?) workflow that I and others have been missing…

> `brew update-reset` can be used to revert to the upstream state on all Homebrew's repositories.

![](https://media.giphy.com/media/11FiDF2fuOujPG/giphy.gif)

**Comment by mcclure** [comment]
Hello! That's neat. However, as noted, I already managed to back out the change in my case. I filed because I do think this is actually a documentation/ux bug. Brew could probably be written to help people out of this small hole when they fall in.

I see this is closed, but could I make a suggestion: Maybe it would be a good idea to mention this `brew update-reset` trick in the `brew help edit` text? Currently `brew update-reset` is not even listed in `brew help`.
